src/cascadia/TerminalSettingsModel/FileUtils.h

```diff
@@ -1,41 +1,11 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-#pragma once
-
 namespace winrt::Microsoft::Terminal::Settings::Model
 {
-    // I couldn't find a wil helper for this so I made it myself
-    class locked_hfile

```

Comment:
??? wait don't we want this? like, a lot?

Reply:
@lhecker?

Reply:
Found it! https://github.com/microsoft/terminal/issues/11357#issuecomment-930264898

Reply:
Notes added to #11357.

Reply:
I think there was another comment somewhere that explained the reasoning for this...
In either case we (maybe me?) decided that we don't need to re-read the file anymore when writing to disk, because we can just move `ApplicationState` into the monarch process instead. This PR doesn't do that yet though...

Reply:
I feel like moving `ApplicationState` into the Monarch process is going to give us significant IPC headaches given that a bunch of things _want_ to use `ApplicationState` locally (command palette, dismissed warnings) and we haven't remoted any of the Terminal model objects yet...

Have we put the cart squarely ahead of the horse here? :smile:

Reply:
For additional context, the thread where it was decided to add this is here: https://github.com/microsoft/terminal/pull/11083#discussion_r715130263